src/uu/dd/src/dd.rs

```diff
@@ -0,0 +1,1296 @@
+        for idx in 0..buf.len() {
+            buf[idx] = ct[buf[idx] as usize];
+        }
+    }
+    // --------------------------------------------------------------------
+    if conv_only(i) {
+        // no block/unblock
+        let ct = i.cflags.ctable.unwrap();
+        apply_ct(&mut buf, ct);
+
+        Ok(buf)
+    } else if should_block_then_conv(i) {
+        // ascii input so perform the block first
+        let cbs = i.cflags.block.unwrap();
+
+        let mut blocks = block(buf, cbs, rstat);
+
+        if let Some(ct) = i.cflags.ctable {
+            for buf in blocks.iter_mut() {
+                apply_ct(buf, ct);
+            }
+        }
+
+        let blocks = blocks.into_iter().flatten().collect();
+
+        Ok(blocks)
+    } else if should_conv_then_block(i) {
+        // Non-ascii so perform the conversion first
+        let cbs = i.cflags.block.unwrap();
+
+        if let Some(ct) = i.cflags.ctable {
+            apply_ct(&mut buf, ct);
+        }
+
+        let blocks = block(buf, cbs, rstat).into_iter().flatten().collect();
+
+        Ok(blocks)
+    } else if should_unblock_then_conv(i) {
+        // ascii input so perform the unblock first
+        let cbs = i.cflags.unblock.unwrap();
+
+        let mut buf = unblock(buf, cbs);
+
+        if let Some(ct) = i.cflags.ctable {
+            apply_ct(&mut buf, ct);
+        }
+
+        Ok(buf)
+    } else if should_conv_then_unblock(i) {
+        // Non-ascii input so perform the conversion first
+        let cbs = i.cflags.unblock.unwrap();
+
+        if let Some(ct) = i.cflags.ctable {
+            apply_ct(&mut buf, ct);
+        }
+
+        let buf = unblock(buf, cbs);
+
+        Ok(buf)
+    } else {
+        // The following error should not happen, as it results from
+        // insufficient command line data. This case should be caught
+        // by the parser before making it this far.
+        // Producing this error is an alternative to risking an unwrap call
+        // on 'cbs' if the required data is not provided.
+        Err(Box::new(InternalError::InvalidConvBlockUnblockCase))
+    }
+}
+
+fn read_helper<R: Read>(
+    i: &mut Input<R>,
+    bsize: usize,
+) -> Result<(ReadStat, Vec<u8>), Box<dyn Error>> {
+    // Local Predicate Fns -----------------------------------------------
+    #[inline]
+    fn is_conv<R: Read>(i: &Input<R>) -> bool {
+        i.cflags.ctable.is_some()
+    }
+    #[inline]
+    fn is_block<R: Read>(i: &Input<R>) -> bool {
+        i.cflags.block.is_some()
+    }
+    #[inline]
+    fn is_unblock<R: Read>(i: &Input<R>) -> bool {
+        i.cflags.unblock.is_some()
+    }
+    // Local Helper Fns -------------------------------------------------
+    #[inline]
+    fn perform_swab(buf: &mut [u8]) {
+        let mut tmp;
+
+        for base in (1..buf.len()).step_by(2) {
+            tmp = buf[base];
+            buf[base] = buf[base - 1];
+            buf[base - 1] = tmp;
```

Comment:
You could use [`swap`](https://doc.rust-lang.org/std/primitive.slice.html#method.swap)

Reply:
I made this change.